readme files for components and tests containing naming convention (#2897)
kyma-project/kyma: tests/README.md

# Tests

## Overview

> The `tests` folder contains the sources for all Kyma tests.

Reviewer: ```suggestion
The `tests` directory contains the sources for all Kyma tests.
```

A Kyma test is Pod/container/image referenced in a Kyma module/chart test section to provide the module's test functionality. 

> A Kyma test gets executed against a running Kyma cluster to assure integrity and functional correctness of the cluster with all modules installed. These are acceptance tests.

Author: You might think that there are other tests as well...

> A Kyma test gets executed against a running Kyma cluster to assure integrity and functional correctness of the cluster with all modules installed. These are acceptance tests.

Author: Can we maybe just remove that small little sentence and that's it. Let's just not mention acceptance tests and we are fine :)

> Each subdirectory in the tests directory defines sources for one test suite, usually focusing on one component. The resulting docker images are then referenced by the related Kyma modules or charts.

Reviewer: ```suggestion
Each subdirectory in the tests directory defines sources for one test suite, usually focusing on one component. The resulting Docker images are then referenced by the related Kyma modules or charts.
```

## Details

> Every Kyma test resides in a dedicated folder containing its sources and a `README.md ` file. This file provides instructions on how to build and develop the test suite.

Reviewer: ```suggestion
Every Kyma test resides in a dedicated folder which contains its sources and a `README.md ` file. This file provides instructions on how to build and develop the test suite.
```

> The test name, which is also the folder name, reflects the tested component without any prefix or suffix. For example,  `monitoring`.

Reviewer: ```suggestion
The test name, which is also the folder name,  is the component's name without any prefix or suffix. For example,  `monitoring`.
```

> The docker image resulting from the sources of a test suite resides in the tests subfolder.

Reviewer: ```suggestion
The Docker image resulting from the sources of a test suite resides in the dedicated `tests` subfolder.
```

> Example: The Event Bus component has its acceptance tests in the `tests/event-bus` folder and produces the `XX/tests/event-bus:0.5.1` docker image.

Reviewer: ```suggestion
Example: The Event Bus component has its acceptance tests in the `tests/event-bus` folder and produces the `XX/tests/event-bus:0.5.1` Docker image.
```
